### PR TITLE
add Posit Cloud PublishProvider

### PR DESCRIPTION
**Pull request by mslynch**
## Description

Adds support for publishing to Posit Cloud by implementing a new `PublishProvider`.

## Checklist

I have (if applicable):

- [x] referenced the GitHub issue this PR closes
- [x] updated the appropriate changelog

### CODE REVIEW

File: src/publish/common/bundle.ts
Comment (mslynch) [inline src/publish/common/bundle.ts:28]: I found that this was calculating the wrong checksum – it only works because Posit Connect doesn't validate it. `md5Hash` works with strings, but I couldn't find a way to get it to work properly using the contents of a file. Since Deno's crypto module supports MD5, there's probably no reason to keep `blueImpMd5` – but I'll leave that to a future PR.

File: src/publish/posit-cloud/posit-cloud.ts
Comment (jjallaire) [inline src/publish/posit-cloud/posit-cloud.ts:124]: I think here and in the subsequent hint you should be directing the user to the exact page where they provision a token and secret for their account. With this link, they get once sentence about tokens, then need to realize that they should click "Publish Outputs", then read some general conceptual material about Publish Outputs, then realize that they need to skip past "Publish from a Cloud Project", then ???. I think this link should land them directly on the provisioning page and if additional explanation is required it should be there inline.

**Comment by jjallaire** [comment]
Implementation looks great! My only 2 comments are:

(1) Previous note about bringing the user directly to the token provisioning page (if you don't have this yet then I'd still put the right URL here and then just redirect for the time being)

(2) I'm guessing you will want to hide this for the time being. The only place the user will see "Posit Cloud" if they aren't directly looking for it is in the interactive list of providers. You can filter Posit Cloud out this list here: https://github.com/quarto-dev/quarto-cli/blob/main/src/command/publish/cmd.ts#L226. Note that users will still be able to publish via `quarto publish posit-cloud`, they just won't see it in the menu.

File: src/publish/posit-cloud/posit-cloud.ts
Comment (mslynch) [reply src/publish/posit-cloud/posit-cloud.ts]: Posit Cloud doesn't have such a URL at the moment, unfortunately... The URL for that page is specific to the user's account. I filed an issue for us to add a generic tokens URL and took out the `hint`.

**Review by jjallaire — CHANGES_REQUESTED** [review]
This is good to go after you make the change to not have posit cloud show up in the providers prompt (at least until after you launch)

**Comment by jjallaire** [comment]
You can't remove it entirely from `publishProviders()` (else it won't work in all cases). You need to filter it out when you call for that to build the prompt UI (see link I sent before here: https://github.com/quarto-dev/quarto-cli/blob/main/src/command/publish/cmd.ts#L226)

**Pull request merged**
